lightning/src/chain/channelmonitor.rs

```diff
@@ -626,7 +632,8 @@ pub struct ChannelMonitor<ChanSigner: ChannelKeys> {
 	counterparty_payment_script: Script,
 	shutdown_script: Script,
 
-	keys: ChanSigner,
+	key_derivation_params: (u64, u64),
```

Comment:
Wouldn't be better in a future move to replace those two values by a ChannelKeys identifier and store them behind the interface to never expose them in-memory ? Those values are consumed in ChannelKeys derivation and might it could be better to keep them private ?

Anyway it would be great to add a one-line comment about their purpose in ChannelMonitor to serve at onchain output description, without constraining the utxo signer to store ChannelKeys.

Reply:
I'm not exactly sure what you're asking - are you suggesting we rename them "channel_keys_id" or similar? While our default channel keys stores actual key derivation info in them, that's certainly not a requirement, nor should key derivation information generally be particularly sensitive.

Reply:
With our current `KeysManager::derive_channel_keys`, those two values are inputs to the commitment seed, which is used to derive per-channel private keys. Theoretically I think an attacker able to observe them in system memory will have an easier task to re-compute the `commitment_seed`. In practice, we're already aware of this limitation as comment in `keysinterface.rs` (L754) underscores it.

If we want to move from this situation, we can have the `KeysManager` provide an _arbitrary_ identifier number to the `derive_channel_keys` caller. This identifier could be stored by `ChannelMonitor` and pass back in any `SpendableOutputDescriptor`. That way, you can have a key holder implementation on an external signing device, it consumes a `SpendableOutputDescriptor`, derives the channel keys and gets back the onchain output, without having ever exposed the derivation params.

I think that's a side-point to the current proposed change, so not a blocker.

Reply:
These values *are* an arbitrary identifier. They happen to be 128 bits in size so that keys interface implementors can use them as cryptographic values if they want, but there's nothing stopping them from being just an ID. The documentation on it can be better, indeed, but I think the API is already fine.